sensor/kubernetes/sensor/sensor.go

```diff
@@ -191,11 +191,7 @@ func CreateSensor(cfg *CreateOptions) (*sensor.Sensor, error) {
 		log.Errorf("Failed to determine namespace from service account token file: %s", err)
 	}
 	if sensorNamespace == "" {
-		sensorNamespace = os.Getenv("POD_NAMESPACE")
-	}
-	if sensorNamespace == "" {
-		sensorNamespace = namespaces.StackRox
-		log.Warnf("Unable to determine Sensor namespace, defaulting to %s", sensorNamespace)
+		sensorNamespace = sensorupgrader.GetSensorNamespace()
```

Comment:
Without an explanation this is very confusing. We have an exported function `GetSensorNamespace()` that aims to be the authoritative source on where Sensor runs. However, here it is used as a fallback. Why? What are the reasons and consequences of having two different mechanisms for determining Sensor namespace?

Reply:
The idea here is to remain consistent with previous behavior in sensor, see my comment above. I could also undo this change and the code would be the same, since `sensorupgrader.GetSensorNamespace()` now queries `POD_NAMESPACE` and defaults on `stackrox` if it's unset.

 I only plugged the function in there to avoid code duplication, but if this is harder to read I can also remove it and leave a little bit of duplicate code in sensor.
 
 > GetSensorNamespace() [...] aims to be the authoritative source on where Sensor runs
 
Yes, however I do not want to inadvertently break / change sensor behavior by no longer querying the `SAToken` namespace in sensor. However querying the `SAToken` namespace in the sensor upgrader led to unexpected behavior that I was not able to fully resolve, so having `GetSensorNamespace` as it is in the PR now is a compromise since the Konflux sprint ended.

Rather then keep the upgrader on a static namespace there is now some capability for supporting other namespaces.

Reply:
Yeah, I understand your motives. I talk about the effect it might have on posterity. What I think we're saying now is:
> There is `GetSensorNamespace()` and we _think_ it's the right way to determine Sensor namespace. However, Malte made it differently and we don't know why but are hesitant to change his approach now.

How I would read the code snippet above if I would stumble across it:
> `GetSensorNamespace()` does not quite work here because this case is special but we won't tell you in what sense.

I see a few options:

1. add a comment here saying we probably should delete the satoken approach
2. add a param to `GetSensorNamespace()` to turn on the satoken approach; this way you keep a single interface for the single task
3. removing satoken approach everywhere.

Reading up comments in https://issues.redhat.com/browse/ROX-6142 and given we're in the beginning of the release cycle my personal preference goes for option 3.